tools/shared.py

```diff
@@ -2944,7 +2949,19 @@ def get_binaryen_bin():
 
   @staticmethod
   def run_binaryen_command(tool, infile, outfile=None, args=[], debug=False, stdout=None):
-    cmd = [os.path.join(Building.get_binaryen_bin(), tool)] + args
+    cmd = [os.path.join(Building.get_binaryen_bin(), tool)]
+    if outfile and tool == 'wasm-opt' and Settings.DEBUG_LEVEL != 3:
+      # remove any dwarf debug info sections, if the debug level is <3, as
+      # we don't need them; also remove them if we the level is 4, as then we
+      # want a source map, which is implemented separately from dwarf.
+      # note that we add this pass first, so that it doesn't interfere with
+      # the final set of passes (which may generate stack IR, and nothing
+      # should be run after that)
+      # TODO: once fastcomp is gone, either remove source maps entirely, or
+      #       support them by emitting a source map at the end from the dwarf,
+      #       and use llvm-objcpy to remove that final dwarf
+      cmd += ['--strip-dwarf']
```

Comment:
If we want to strip early why not just have the linker not emit the debug into at all?

Reply:
oh, how can we do that? does lld have a flag?

Reply:
I see there is `--strip-debug`. Does that remove both dwarf and the names section, though? here we only want to remove dwarf.

Reply:
Yes it will strip the name section too.  Maybe add a comment about this here?  We could potentially add a wasm-ld for controlling the name section separately .. but its not there today.

Reply:
Good idea, added a comment now.